documentation/docs/specs/revocation_bitmap_2022.md

```diff
@@ -0,0 +1,182 @@
+
+## Revocation Bitmap Concept
+
+The revocation status of a verifiable credential is expressed as a binary value. The issuer keeps a bitmap of indices corresponding to verifiable credentials it has issued. If the binary value of the index in the bitmap is 1 (one), the verifiable credential is revoked, if it is 0 (zero) it is not revoked.
+
+## Data Model
+
+### Revocation Bitmap Status
+
+For an issuer to enable verifiers to check the status of a verifiable credential, the [`credentialStatus`](https://www.w3.org/TR/vc-data-model/#status) property MUST be specified with the following properties:
+
+- **id**: A [DID URL](https://www.w3.org/TR/did-core/#did-url-syntax) that can be resolved to a service of the issuer which contains the revocation bitmap;
+- **type**: the string `"RevocationBitmap2022"`;
+- **revocationBitmapIndex**: an unsigned, 32-bit integer expressed as a string, which is the index of the verifiable credential's revocation status in the issuer's revocation bitmap;
+
+### Revocation Bitmap Service
+
+In order to allow verfiers to check the status of a verifiable credential, the DID document of the issuer may contain a service that MUST have the following properties:
+
+- **id**: a URL identifying the revocation bitmap;
+- **type**: the string `"RevocationBitmap2022"`;
+- **serviceEndpoint**: a `data` URL [[RFC2397](https://datatracker.ietf.org/doc/html/rfc2397)] embedding a revocation bitmap generated according to the [bitmap generation algorithm](#Bitmap-Generation-Algorithm). The `<mediatype>` MUST be `application/octet-stream` and the `base64` attribute MUST be set.
+
+## Algorithms
+
+The following algorithms describe how to generate and validate revocation bitmaps.
+
+### Bitmap Generation Algorithm
+
+The following process MUST be followed when producing a `RevocationBitmap2022`:
+
+1. Let **bitmap** be a [_roaring bitmap_](https://roaringbitmap.org/) where each bit is initialized to 0 (zero).
+2. For each revoked credential with an **identifier** not exceeding an unsigned, 32-bit integer, set the bit in **bitmap** at index **identifier** to 1.
+3. Generate the **bitmap serialization** according to the [roaring bitmap serialization format](https://github.com/RoaringBitmap/RoaringFormatSpec/) using the **bitmap** as input.
+4. Generate a **compressed bitmap** by using the ZLIB compression algorithm [[RFC 1950](https://datatracker.ietf.org/doc/html/rfc1950)] on the bitmap serialization and base64-encoding [[RFC4648](https://datatracker.ietf.org/doc/html/rfc4648)] the result.
+5. Return the **compressed bitmap**.
+
```

Comment:
Isn't this missing a step to produce the `Revocation Bitmap Service` itself? I.e. encoding it in the data URL and placing it in the service endpoint? RevocationList2022 has two separate algorithms for the separate steps, but we only include one: https://w3c-ccg.github.io/vc-status-rl-2020/#generate-algorithm

Reply:
Yes, good point. I renamed those two sections to service endpoint generation/expanstion algorithm, please take look if it's better now.